src/src/PluginStructs/P121_data_struct.cpp

```diff
@@ -0,0 +1,48 @@
+#include "../PluginStructs/P121_data_struct.h"
+
+#ifdef USES_P121
+
+// Needed also here for PlatformIO's library finder as the .h file
+// is in a directory which is excluded in the src_filter
+#include <Adafruit_Sensor.h>
+#include <Adafruit_HMC5883_U.h>
+
+bool P121_data_struct::begin(int taskid)
+{
+  if (!initialized)
+  {
+    mag = Adafruit_HMC5883_Unified(taskid);
+    initialized = mag.begin();
+
+    if (initialized)
+    {
+      // Set up oversampling and filter initialization
+      sensor_t sensor;
+      mag.getSensor(&sensor);
+      #ifndef BUILD_NO_DEBUG
+      if (loglevelActiveFor(LOG_LEVEL_DEBUG))
+      {
+        addLog(LOG_LEVEL_DEBUG, F("------------------------------------"));
```

Comment:
Adding these lines in log output isn't really helpful, IMHO, you could start the log with `HMC5883L`.
And also the use of spaces in log messages should better be minimized to save some binary size (and let the compiler de-duplicate strings more easy), log is not about pretty, but about functional.

Reply:
Yes, agree. I was stuck to stright to the example code of Adafruit.